salt/states/git.py

```diff
@@ -721,6 +724,31 @@ def latest(name,
                 )
                 local_changes = False
 
+            if local_changes:
```

Comment:
Yeah, I did it this way because I was planning on doing something if revs_match wasn't ``True``, but I decided not to. This can be put into a single conditional. I'm also going to look at the tests, I think this PR will actually cause one of them to fail, so the test needs to be adjusted.